### ISSUE TITLE
Allow customization for disabling View.isInEditMode

### ISSUE THREAD

**Issue by elihart** [cashapp/paparazzi#614]
The PR that forced isInEditMode to false (https://github.com/cashapp/paparazzi/pull/145) assumes that people will want that behavior. We actually want our `isInEditMode` to be respected in screenshots, mostly so that we know when not to access the Application object and dagger graph (mostly within low level infra such as image views and logging).

Can a parameter be added to Paparazzi to control whether the method interceptor for View.isInEditMode is added?

**Comment by jrodbx** [comment]
Paparazzi is of the opinion that you should never want to do this.  That's one key differentiator between Layoutlib and Paparazzi.  Paparazzi wants you to run your actual code!

Also, 3rd party or framework views might package logic gated by isInEditMode.

Might I suggest a lint check enforcing the Application object access, and refactoring the Dagger graph out of your views and into some presenter/controller/manager object and using fakes to drive test code?

Closing as wontfix.

**Comment by elihart** [comment]
I see that that is ideal, but it isn't feasible for us to make a refactor that large. For example, we aren't going to refactor every image view usage to pass the Glide instance down from the Fragment. We are only using Paparazzi for new compose screens, but when developers embed old views via interop we run into these issues and we need an escape hatch to make paparazzi work. It's unfortunate that you won't consider adding this flexibility as otherwise we'll need to maintain a fork to be able to use paparazzi

**Comment by elihart** [comment]
Is there any non-ideological reason to enforce this? What would be wrong with defaulting to your opinionated approach while allowing an escape hatch some how?
